examples/hpc/dla-driver/src/lib.rs

```diff
@@ -285,28 +297,82 @@ impl Dla {
     }
 
     /// Reads len amount of bytes from DLA's output bank(s)
-    pub fn read_output(&self, len: usize) -> Vec<i8> {
-        // VP only support reading from banks
-        if cfg!(feature = "vp") {
-            return self.read_data_bank(self.get_output_bank(), len);
+    pub fn read_output_i32(&self, len: usize) -> Vec<i32> {
+        let bytes = self.read_data_bank(self.get_output_bank(), len * 4);
+        let mut result = Vec::with_capacity(bytes.len() / 4);
+
+        for pair in bytes.chunks_exact(4) {
+            let combined = ((pair[0] as i32) << 24)
+                | ((pair[1] as i32) << 16)
+                | ((pair[2] as i32) << 8)
+                | (pair[3] as i32);
+            result.push(combined);
+        }
+        result
+    }
+
+    /// Reads len amount of bytes from DLA's output bank(s)
+    pub fn read_output_i16(&self, len: usize) -> Vec<i16> {
+        let bytes = self.read_data_bank(self.get_output_bank(), len * 2);
+        let mut result = Vec::with_capacity(bytes.len() / 2);
+
+        for pair in bytes.chunks_exact(2) {
+            let combined = ((pair[0] as i16) << 8) | (pair[1] as i16 & 0xFF);
+            result.push(combined);
+        }
+        result
+    }
+
+    /// Reads len amount of bytes from DLA's output bank(s)
+    pub fn read_output_i8(&self, len: usize) -> Vec<i8> {
+        let bytes = self.read_data_bank(self.get_output_bank(), len);
+        bytes.iter().map(|&x| x as i8).collect()
+    }
+
+    /// Reads len amount of bytes from DLA's output bank(s)
+    pub fn read_output_i4(&self, len: usize) -> Vec<i8> {
```

Comment:
This is used to read and unpack 4-bit SIMD calculations from DLA. In short it's possible to do calculation on DLA by packing two or four values to each byte by reducing the bit width of input and kernel values. This allows for more parallelism with the cost of granularity in results. My idea here was to unpack each byte to two i8's since as per my understanding i4 and i2 types aren't supported.